packages/blitz-dom/src/events/mouse.rs

```diff
@@ -275,4 +287,30 @@ pub(crate) fn handle_click<F: FnMut(DomEvent)>(
 
     // If nothing is matched then clear focus
     doc.clear_focus();
+
+    // Assumed double click time to be less than 500ms, although may be system-dependant?
+    if doc.dbl_click_first_time.map(|t| t.elapsed() < Duration::from_millis(500)).unwrap_or(false)  {
+        dispatch_event(DomEvent::new(target, DomEventData::DoubleClick(event.clone())));
+        doc.dbl_click_first_time = None;
+    } else {
+        doc.dbl_click_first_time = Some(Instant::now());
+    }
 }
```

Comment:
OS's do have a system setting for this. But I think assuming 500ms is fine for now, we can add that later.

What I think we will want is to generalise the "double click" tracking into a `click_count: u16` and `last_click_time`. Where we keep incrementing the count while clicks are still within the time limit (from `last_click_time`) and "double click" corresponds to click count 2.

Try clicking these examples many times in Chrome/Firefox:

- https://developer.mozilla.org/en-US/docs/Web/API/Element/dblclick_event#result
- https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event#result

Reply:
Non-blocking, but we may want to consider generating `Instant::now()` in `blitz-shell` and have `blitz-dom`'s event handling functions take it as a parameter (/make it a field in the event struct). This would be really useful for testing things like click detection, and would also allow us to make use of event timestamps provided by the OS if/when those are available from Winit.